File: server/tests/fixtures/alpha_rule_triggers/typescript/architecture_critical_dependency.ts
// architecture.critical_dependency fixture
// This file is a critical dependency bottleneck - many modules depend on it
// Should trigger: architecture.critical_dependency

export class CriticalDependency {
    coreOperation(data: any): object {
        return this.process(data);
    }
    
    private process(data: any): object {
        return { result: data };
    }
}

export function criticalUtility(): CriticalDependency {
    return new CriticalDependency();
}

File: server/tests/fixtures/alpha_rule_triggers/python/architecture_critical_dependency.py
# architecture.critical_dependency fixture
# This file is a critical dependency bottleneck - many modules depend on it
# Should trigger: architecture.critical_dependency

class CriticalDependency:
    """A class that represents a critical architectural dependency."""
    
    def core_operation(self, data):
        """Core operation that many modules use."""
        return self._process(data)
    
    def _process(self, data):
        return {"result": data}

def critical_utility():
    """Utility function that is widely used."""
    return CriticalDependency()
